source/grid/grid_generator.cc

```diff
@@ -725,26 +728,111 @@ namespace GridGenerator
     tria.set_manifold(0, TorusManifold<2>(R, r));
   }
 
+
+
   template <>
-  void torus<3, 3>(Triangulation<3, 3> &tria, const double R, const double r)
+  void torus<3, 3>(Triangulation<3, 3> &tria,
+                   const double         R,
+                   const double         r,
+                   const unsigned int   n_cells_toroidal)
   {
     Assert(R > r,
            ExcMessage("Outer radius R must be greater than the inner "
                       "radius r."));
     Assert(r > 0.0, ExcMessage("The inner radius r must be positive."));
+    Assert(n_cells_toroidal > 2,
+           ExcMessage("Number of cells in toroidal direction has "
+                      "to be at least 3."));
+
+    // the first 8 vertices are in the x-y-plane
+    Point<3> const        p = Point<3>(R, 0.0, 0.0);
+    double const          a = 1. / (1 + std::sqrt(2.0));
+    std::vector<Point<3>> vertices(8 * n_cells_toroidal);
+    vertices[0] = p + Point<3>(-1, -1, 0) * (r / std::sqrt(2.0)),
+    vertices[1] = p + Point<3>(+1, -1, 0) * (r / std::sqrt(2.0)),
+    vertices[2] = p + Point<3>(-1, -1, 0) * (r / std::sqrt(2.0) * a),
+    vertices[3] = p + Point<3>(+1, -1, 0) * (r / std::sqrt(2.0) * a),
+    vertices[4] = p + Point<3>(-1, +1, 0) * (r / std::sqrt(2.0) * a),
+    vertices[5] = p + Point<3>(+1, +1, 0) * (r / std::sqrt(2.0) * a),
+    vertices[6] = p + Point<3>(-1, +1, 0) * (r / std::sqrt(2.0)),
+    vertices[7] = p + Point<3>(+1, +1, 0) * (r / std::sqrt(2.0));
+
+    // create remaining vertices by rotating around positive y-axis
```

Comment:
Could you change the comment to "negative y-axis"? Originally, I used a rotation around positive y-axis but this gives a wrong numbering of vertices (negative cell measure) and then forgot to update the comment. A further explanation "... to ensure positive cell measure" could also be helpful.

Reply:
Done!